server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -3623,7 +3623,11 @@ public NicSecondaryIpResponse createSecondaryIPToNicResponse(NicSecondaryIp resu
         NicVO nic = _entityMgr.findById(NicVO.class, result.getNicId());
         NetworkVO network = _entityMgr.findById(NetworkVO.class, result.getNetworkId());
         response.setId(result.getUuid());
-        response.setIpAddr(result.getIp4Address());
+        if (result.getIp4Address() != null) {
```

Comment:
would you mind extracting the block 3626-3630 to a method? You can then unit test it.
Moreover, it looks like this block is also used at  3713-3717

Reply:
Done. Thanks, @rafaelweingartner.